src/main/java/duke/misc/TaskList.java

```diff
@@ -0,0 +1,152 @@
+package duke.misc;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import duke.exception.InvalidDataException;
+import duke.exception.InvalidDescriptionException;
+import duke.exception.InvalidIndexException;
+import duke.exception.InvalidTypeException;
+import duke.task.Task;
+
+/**
+ * TaskList class to store all the tasks in Duke.
+ */
+public class TaskList {
```

Comment:
Maybe you can consider the Command pattern to handle the user commands, so some of the string output responsibilities can be abstracted out?